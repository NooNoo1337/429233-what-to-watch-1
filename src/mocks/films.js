export default [
  {
    id: 1,
    title: `Fantastic Beasts`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: ``,
    src: ``
  },
  {
    id: 2,
    title: `Bohemian Rhapsody`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: ``,
    src: ``
  },
  {
    id: 3,
    title: `Macbeth`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: ``,
    src: ``
  },
  {
    id: 4,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    title: `Aviator`,
    genre: ``,
    src: ``
  },
  {
    id: 5,
    title: `Filth`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: ``,
    src: ``
  },
];
