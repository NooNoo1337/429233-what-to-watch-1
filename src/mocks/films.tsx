import {Film} from "../types";

const mockFilms: Film[] = [
  {
    'id': 1,
    'name': `The Grand Budapest Hotel`,
    'poster_image': `img/the-grand-budapest-hotel-poster.jpg`,
    'preview_image': `img/the-grand-budapest-hotel.jpg`,
    'background_image': `img/the-grand-budapest-hotel-bg.jpg`,
    'background_color': `#ffffff`,
    'video_link': `https://some-link`,
    'preview_video_link': `https://some-link`,
    'description': `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    'rating': 8.9,
    'scores_count': 240,
    'director': `Wes Andreson`,
    'starring': [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    'run_time': 99,
    'genre': `Comedy`,
    'released': 2014,
    'is_favorite': false,
  }
];

const mockGenres = [`All genres`, `Horror`, `Drama`];

const mockFilmsToShow = 20;

const mockParams = {params: {id: 1}};

const mockAccountData = {
  avatar_url: `/wtw/static/avatar/9.jpg`,
  email: `test@test.ru`,
  id: 1,
  name: `test`
};

const mockComments = [
  {
    id: 1,
    user: {
      id: 4,
      name: "Kate Muir",
    },
    rating: 8.9,
    comment: "Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.",
    date: "2019-05-08T14:13:56.569Z"
  }
];

export {
  mockFilms,
  mockGenres,
  mockFilmsToShow,
  mockAccountData,
  mockParams,
  mockComments
};
