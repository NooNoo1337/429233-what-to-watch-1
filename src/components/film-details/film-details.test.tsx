import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import {Provider} from 'react-redux';
import {createStore} from 'redux';

// Components
import FilmDetails from './film-details';

// Reducers
import combineReducers from '../../reducer/index';
const store = createStore(
  combineReducers
);

// Mocks
import {mockFilms} from '../../mocks/films';
const mockParams = {params: {id: 1}};

describe(`FilmDetails`, () => {
  it(`should render correctly`, () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <FilmDetails
              films={mockFilms}
              isPlayerActive={false}
              match={mockParams}
            />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
