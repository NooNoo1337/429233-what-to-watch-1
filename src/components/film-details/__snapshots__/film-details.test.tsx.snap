// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmDetails should render correctly 1`] = `null`;
